### PR TITLE
implemented Xcode license check for `pod setup`

### LINKED ISSUE #3002: Won't running pod setup with sudo create some root-owned stuff in my user-dir?
```
$ pod setup

"Agreeing to the Xcode/iOS license requires admin privileges, please re-run as root via sudo."
```

This seems like a bad suggestion to me. Won't it create a ~/.cocoapods that is owned by root?

In my case, I just launched Xcode and agreed to the ULA, and then re-ran pod setup.

### PR DESCRIPTION
**Pull request by CalQL8ed-K-OS**
fixes #3002 

reviews
- [ ] @kylef 
- [ ] @neonichu 
- [ ] @grav

it's my first PR, so please offer any advice and suggestions you might have.
although functional, this probably could use some refactoring.
i also had no idea how to write a test for this, especially since i don't currently have a machine where i haven't accepted the license agreement.

### CODE REVIEW

**Comment by segiddins** [comment]
Please see your test failures at https://travis-ci.org/CocoaPods/CocoaPods/jobs/48223521#L2177.

**Comment by CalQL8ed-K-OS** [comment]
i was in the processing of installing rubocop to check, didn't expect someone to response that quick, props :)
addressed the issues just now.

**Comment by CalQL8ed-K-OS** [comment]
¡hurray, passing tests!
locally i'm not getting passing tests though.
¿can anyone help me figure out what's wrong locally?
[Output of `bundle exec rake spec`](https://gist.github.com/CalQL8ed-K-OS/248732d2b4004de4d7d5)
it seems to just be a bunch of _undefined method_ errors on __ Pod::SourcesManager:Class__

**Comment by kylef** [comment]
I don't think this should run only for `pod setup`, but instead for every invocation of `pod` (unless is slows down CocoaPods noticeably).

**Comment by segiddins** [comment]
`xcrun clang  0.00s user 0.01s system 29% cpu 0.037 total`

**Comment by CalQL8ed-K-OS** [comment]
looks like it's speed enough.
¿where should i move the method and method call to preempt any pod command?
¿anyone got any advice on my local test failures?

**Comment by kylef** [comment]
@CalQL8ed-K-OS Hey, I think somewhere around https://github.com/CocoaPods/CocoaPods/blob/master/lib/cocoapods/command.rb#L43 would be best.

File: lib/cocoapods/command/setup.rb [hunk 1]
```diff
@@ -44,6 +46,26 @@ def run
         UI.puts 'Setup completed'.green
       end
 
+      def check_xcode_license_approved
+        unless xcode_license_approved?
+          UI.puts xcode_license_message.red
```
Comment (segiddins) [inline lib/cocoapods/command/setup.rb]: `raise Informative, xcode_license_message` ?

Comment (CalQL8ed-K-OS) [inline lib/cocoapods/command/setup.rb]: - ¿do you propose adding that line, or replacing the `UI.puts` call?
- ¿what's the ramifications of that line? ¿who catches it?

**Comment by CalQL8ed-K-OS** [comment]
@kylef , i've implemented the check for all commands now i believe.

i still have local failures when running `rake spec`.
at first glance they all seem to share the error message `NoMethodError: undefined method 'find_or_create_source_with_url' for Pod::SourcesManager:Class`.

i also still have no idea how to really test that this feature works since i don't have a machine without the license accepted.

File: lib/cocoapods/command.rb [hunk 2]
```diff
@@ -121,5 +122,25 @@ def verify_lockfile_exists!
         raise Informative, "No `Podfile.lock' found in the project directory, run `pod install'."
       end
     end
+
+    def self.verify_xcode_license_approved!
+      unless xcode_license_approved?
+        UI.puts xcode_license_message.red
```
Comment (segiddins) [inline lib/cocoapods/command.rb]: `raise Informative, xcode_license_message`

Comment (CalQL8ed-K-OS) [inline lib/cocoapods/command.rb]: i'm not sure about this, it makes for uglier output in my opinion.

### neat
screen shot 2015-02-13 at 10 00 36 am


### ugly
screen shot 2015-02-13 at 10 05 38 am

Comment (orta) [inline lib/cocoapods/command.rb]: Yeah,  I prefer @CalQL8ed-K-OS it's not an assertion where someone needs the additional stack trace

Comment (segiddins) [inline lib/cocoapods/command.rb]: It only shows the backtrace in development mode...

Comment (orta) [inline lib/cocoapods/command.rb]: Ah, then probably better to have it in then :dancers:

Comment (CalQL8ed-K-OS) [inline lib/cocoapods/command.rb]: ahha @segiddins, that's what i was missing :)
will change now

**Comment by CalQL8ed-K-OS** [comment]
just tried testing this on a fresh machine, but i can't.
you need to accept the license to do almost anything from the command line in the Rainforest setup process, including `git clone`

File: lib/cocoapods/command.rb [hunk 3]
```diff
@@ -121,5 +122,24 @@ def verify_lockfile_exists!
...
+        raise Informative, xcode_license_message
+      end
+    end
+
+    def self.xcode_license_approved?
+      !(`/usr/bin/xcrun clang 2>&1` =~ /license/ && !$?.success?)
+    end
+
+    def self.xcode_license_message
+      <<-EOS.strip_heredoc
```
Comment (segiddins) [inline lib/cocoapods/command.rb]: why all the newlines in here?

Comment (CalQL8ed-K-OS) [inline lib/cocoapods/command.rb]: just making the message easier to read IMO

Comment (kylef) [inline lib/cocoapods/command.rb]: Messages are already word wrapped when displayed so this is unnecessary.

Comment (CalQL8ed-K-OS) [inline lib/cocoapods/command.rb]: okay, i don't agree, but i'll yield that over, no biggie.

**Comment by segiddins** [comment]
Thanks so much @CalQL8ed-K-OS!

**Comment by orta** [comment]
:star2:

**Pull request merged**